cfme/tests/openstack/test_host_power_control.py

```diff
@@ -0,0 +1,53 @@
+import pytest
+from utils import testgen
+from cfme.web_ui import Quadicon
+from cfme.infrastructure.host import Host
+from cfme.web_ui import InfoBlock, toolbar
+from cfme.fixtures import pytest_selenium as sel
+from utils.wait import wait_for
+
+pytest_generate_tests = testgen.generate(testgen.provider_by_type,
+                                         ['openstack-infra'], scope='module')
+
+
+def refresh_host():
+    toolbar.select("Configuration", "Provide Nodes", invokes_alert=True)
+
+
+def is_host_on(node):
+    return node.get_detail("Properties", "Power State") == 'on'
+
+
+@pytest.mark.usefixtures("setup_provider_modscope")
+def test_host_power_off(provider):
+    provider.load_details()
+    sel.click(InfoBlock.element("Relationships", "Nodes"))
+    my_quads = list(Quadicon.all())
+    assert len(my_quads) > 0
+    quad = my_quads[0]
+    host = Host(name=quad.name)
+    if is_host_on(host):
+        host.power_off()
+        refresh_host()
+    else:
+        pytest.skip("Node is already off")
+    wait_for(lambda: not is_host_on(host), fail_func=lambda: sel.refresh(),
+             delay=15, timeout="10m")
```

Comment:
I think you should reconsider the logic here. You're waiting for a specific state - off - and if Power State is not 'on', that does not mean that it is 'off'.  Your assert after this wait_for may not even be necessary since the condition for the assert is the same as the condition for the wait. If is_host_on() never returns false, the test fails here with an exception from wait_for. If it does, the wait_for completes and then you're asserting the same thing again. If the wait_for() doesn't raise an exception, the test is done. 

It looks like there is a need for power state information from the host object, and these is_host_on()/is_host_off() methods should be in the Host class so they're available outside of this test module. See cfme.common.vm as an example, where the VM class provides some methods for power control and waiting for specific power states. In VM classes specific to a type of provider we can define and override the unique power states available - ultimately we could have the same structure here for openstack/azure/gce/vsphere/rhev 'host' classes. 

Also, gshefer, I think this timeout is capturing the time it takes for the node to shutdown, for that status to be collected by CFME, and then for it to get refreshed and displayed in the UI, not just the provider/node refresh time. So the timeout here should be several minutes, and if you would like to detect a host/provider refresh failure here it should be handled differently.